### ISSUE TITLE
Typo in .htaccess FilesMatch line for .orig and .save files

### ISSUE THREAD

**Issue by tkcent** [backdrop/backdrop-issues#6334]
## Description of the bug

In the FilesMatch line, missing "|" to separate .php.orig and .php.save file extensions that should be denied.

## Steps To Reproduce

To reproduce the behavior:

1. Create file with .php.orig or .php.save extension in the root directory.
2. Access the file via the web browser

## Actual behavior

File gets downloaded. 

## Expected behavior

Should get 404 error.

## Additional information

Add any other information that could help, such as:
- Backdrop CMS version: v1.26.2, but looks like it goes back to about v1.05
- Web server and its version: Applicable for Apache, but also any other web server where the .htaccess rules have been copied to the web server configuration.

**Comment by argiepiano** [comment]
@tkcent can you please add a note as to why this was closed?

**Comment by kiamlaluno** [comment]
The line currently used is the following one.

```none
<FilesMatch "\.(engine|inc|info|install|make|module|profile|test|po|sh|.*sql|theme|tpl(\.php)?|xtmpl)(~|\.sw[op]|\.bak|\.orig|\.save)?$|^(\.(?!well-known).*|Entries.*|Repository|Root|Tag|Template)$|^#.*#$|\.php(~|\.sw[op]|\.bak|\.orig\.save)$">
```

There is a pipe character between `\.orig` and `\.save`.

**Comment by tkcent** [comment]
@kiamlaluno Yes, that's the typo.  That section is meant to be repeated to hide backup files that are generated when a file is edited.  The first instance only affects .tpl.php files, but not just .php files.

**Comment by kiamlaluno** [comment]
@tkcent Then, this issue should have not been closed, should not it?

**Comment by tkcent** [comment]
Thanks @quicksketch for the security review and reopening this issue.  I have some other suggestions with regards to FilesMatch, but I'll open a separate issue since that is more of a feature request.

**Comment by quicksketch** [comment]
From emails @tkcent sent to us, I think there was concern that this issue should perhaps not be handled publicly, and that's why it was closed. Only repository administrators can fully delete issues.

However, the file-blocking in .htaccess is a security hardening, not an inherent security vulnerability and so it can be handled here. In the past, new extensions have been added in public, and occasionally errors are found in the regex (such as Drupal issues https://www.drupal.org/node/3327115, https://www.drupal.org/node/2508666, or https://www.drupal.org/node/2779833). So I think it is fine that we correct the issue publicly as well.

**Comment by tkcent** [comment]
Created pull request: https://github.com/backdrop/backdrop/pull/4603

**Comment by quicksketch** [comment]
Thanks @tkcent. I confirmed that `.php.orig` and `.php.save` files are now blocked on Apache environments. I merged https://github.com/backdrop/backdrop/pull/4603 into 1.x and 1.26.x. Thanks!
